sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,747 @@
-            config.user_agent_policy,
             config.headers_policy,
-            BearerTokenCredentialPolicy(credentials),
+            config.user_agent_policy,
+            config.proxy_policy,
+            _BearerTokenCredentialPolicy(credentials),
             config.redirect_policy,
             config.retry_policy,
             config.logging_policy,
         ]
-        self._pipeline = Pipeline(transport, policies=policies)
-        models = {
-            "DeletedKey": DeletedKey,
-            "DeletedKeyItem": DeletedKeyItem,
-            "DeletedKeyItemPaged": DeletedKeyItemPaged,
-            "JsonWebKey": JsonWebKey,
-            "Key": Key,
-            "KeyAttributes": KeyAttributes,
-            "KeyCreateParameters": KeyCreateParameters,
-            "KeyItem": KeyItem,
-            "KeyItemPaged": KeyItemPaged,
-        }
-        self._deserialize = Deserializer(models)
-        self._serialize = Serializer(models)
-
-    def create_key(self, name, key_type, size=None, key_ops=None, attributes=None, tags=None, curve=None, **kwargs):
-        # type: (str, str, Optional[int], Optional[List[str]], Any, Any, Any, Any) -> Key
-        url = "/".join([self.vault_url, "keys", name, "create"])
-        headers = {"Content-Type": "application/json; charset=utf-8", "x-ms-client-request-id": str(uuid.uuid1())}
-        create_params = KeyCreateParameters(
-            kty=key_type, key_size=size, key_ops=key_ops, key_attributes=attributes, tags=tags, curve=curve, **kwargs
+        transport = RequestsTransport(config)
+        pipeline = Pipeline(transport, policies=policies)
+
+        self._client = KeyVaultClient(credentials, api_version=self._api_version, pipeline=pipeline)
+
+    @property
+    def vault_url(self):
+        # type: () -> str
+        return self._vault_url
+
+    def _create_key(
+        self,
+        name,
+        key_type,
+        size=None,
+        key_ops=None,
+        enabled=None,
+        expires=None,
+        not_before=None,
+        tags=None,
+        curve=None,
+        **kwargs
+    ):
+        # type: (str, str, Optional[int], Optional[List[str]], Optional[bool], Optional[datetime], Optional[datetime], Optional[Dict[str, str]], Optional[str], Mapping[str, Any]) -> Key
+        if enabled is not None or not_before is not None or expires is not None:
+            attributes = self._client.models.KeyAttributes(enabled=enabled, not_before=not_before, expires=expires)
+        else:
+            attributes = None
+
+        bundle = self._client.create_key(
+            self.vault_url, name, key_type, size, key_attributes=attributes, key_ops=key_ops, tags=tags, curve=curve
         )
-        body = self._serialize.body(create_params, "KeyCreateParameters")
-        request = HttpRequest("POST", url, headers)
-        request.set_json_body(body)
-        request.format_parameters({"api-version": self.API_VERSION})
+        return Key._from_key_bundle(bundle)
+
+    def create_key(
```

Comment:
So this allows specifying `key_type` but other important particulars, like size or curve, will be defaults? Is this how other languages have implemented it? Maybe I'm missing something here. To my eyes, `_create_key` looks like a more natural `create_key` because it allows setting everything.

Reply:
Yes. `create_key` uses the `key_type` attribute and defaults the particulars of size and curve and we wanted an implementation that would allow a user to just specify the `key_type` without having to deal with any of the other specifics. Hence this method.

This is a teams' conversation, you can find more info [here](https://teams.microsoft.com/l/message/19:eed5a67bbf6649238d7f802f324aa2ea@thread.skype/1558625285789?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1558625285789&teamName=Azure%20SDK&channelName=Service%20-%20KeyVault&createdTime=1558625285789)

Reply:
That's about adding create overloads in the more typey languages:
1. `CreateKey(string name, KeyType keyType)`: create a key specifying only name and type. Creates a key with default size/curve and metadata.
2. `CreateKey(KeyCreateOptions options)`: create a key specifying name, type, not_before, expires, tags (fields of `KeyCreateOptions`). Creates a key with default size/curve but non-default metadata.

We already have those in `_create_key`. It only requires `name` and `type`. Method 1 is `_create_key(name, "RSA")`. Method 2 is `_create_key(name, "RSA", not_before=now, expires=tomorrow, tags={"spam": "eggs"})`.